json4s/src/main/scala/org/http4s/json4s/Json4sInstances.scala

```diff
@@ -7,27 +7,28 @@
 package org.http4s
 package json4s
 
-import cats.effect.Sync
+import cats.effect.Concurrent
 import cats.implicits._
 import org.http4s.headers.`Content-Type`
 import org.json4s._
 import org.json4s.JsonAST.JValue
 import org.typelevel.jawn.support.json4s.Parser
 
+import scala.util.Try
+
 object CustomParser extends Parser(useBigDecimalForDouble = true, useBigIntForLong = true)
 
 trait Json4sInstances[J] {
-  implicit def jsonDecoder[F[_]](implicit F: Sync[F]): EntityDecoder[F, JValue] =
+  implicit def jsonDecoder[F[_]](implicit F: Concurrent[F]): EntityDecoder[F, JValue] =
     jawn.jawnDecoder(F, CustomParser.facade)
 
-  def jsonOf[F[_], A](implicit reader: Reader[A], F: Sync[F]): EntityDecoder[F, A] =
+  def jsonOf[F[_], A](implicit reader: Reader[A], F: Concurrent[F]): EntityDecoder[F, A] =
     jsonDecoder.flatMapR { json =>
       DecodeResult(
-        F.delay(reader.read(json))
-          .map[Either[DecodeFailure, A]](Right(_))
-          .recover { case e: MappingException =>
-            Left(InvalidMessageBodyFailure("Could not map JSON", Some(e)))
-          })
+        F.pure(
+          Try(reader.read(json)).toEither
+            .leftMap[DecodeFailure](e => InvalidMessageBodyFailure("Could not map JSON", Some(e)))
+        ))
```

Comment:
Yes, `F.pure` is correct here.  I think the old `delay` was capturing exceptions.

This could be microoptimized and avoid the allocation of a `Try` with a regular old `try`/`catch` expression.

Reply:
Thx for the feedback.
I was not sure if a good old `try` / `catch` block was an ok-style in https.
I'll change that.

Reply:
https://github.com/http4s/http4s/pull/3885/commits/1497f7d75faf92e8bb59799ca9f938e832af9d7f